### PR TITLE
#786 added one example in challenge section

### PR DESCRIPTION
**Pull request by Kumaranupam2020**
PR #786 
recursion make it quite easy to calculate gcd of two numbers.I added one challenge of GCD with full explanation . under issue no. #570

### CODE REVIEW

**Comment by manaswinidas** [comment]
Please mention the issue number in the description. It makes your PR easy for evaluation

File: recursion/recursion.md [hunk 1]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
 
 `fun1(1, 1)`
-`fun1(2, 1)`
+`fun1(2, 1)` 
 `fun1(2, 2)`
 `fun1(0, 2)`
+
+#### [Challenge] - GCD
+
+The [GCD] (https://en.wikipedia.org/wiki/Greatest_common_divisor)of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
```
Comment (praveenscience) [inline recursion/recursion.md]: You need to remove the space or else, this won't be appearing as a link.

File: recursion/recursion.md [hunk 2]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+The greatest common divisor(GCD) is also known as:-
```
Comment (praveenscience) [inline recursion/recursion.md]: No need of a `-` at the end.

File: recursion/recursion.md [hunk 3]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+* highest common divisor.
+
+
+For example:-
+
+the gcd of 48 and 14 is 2.
+
+### pseudocode to calculate gcd of two numbers:
+
+GCD(x, y)
```
Comment (praveenscience) [inline recursion/recursion.md]: Enclose this inside <code>```</code> to make it look like a code.

File: recursion/recursion.md [hunk 4]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+
+The [GCD] (https://en.wikipedia.org/wiki/Greatest_common_divisor)of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
+The greatest common divisor(GCD) is also known as:-
+* the greatest common factor (gcf),
+* highest common factor (hcf),
+* greatest common measure (gcm),
+* highest common divisor.
+
+
+For example:-
```
Comment (praveenscience) [inline recursion/recursion.md]: Remove the `-` at the end.

File: recursion/recursion.md [hunk 5]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+
+the gcd of 48 and 14 is 2.
```
Comment (praveenscience) [inline recursion/recursion.md]: Capitalise GCD.

File: recursion/recursion.md [hunk 6]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+          return x;
+      else
+          Call: GCD(y, x%y);
+      endif
+End
+
+### Check for understanding
+
+Finding GCD of 48 & 14 :
+14 ) 48 ( 3
```
Comment (praveenscience) [inline recursion/recursion.md]: Enclose this inside <code>```</code> to make it look like a code. Add a new line before this line.

File: recursion/recursion.md [hunk 7]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+     42
+     -----
+      6
+
+Remainder is not yet zero, so we will now divide 14 by 6
+
+14 ) 48 ( 3
```
Comment (praveenscience) [inline recursion/recursion.md]: Enclose this inside <code>```</code> to make it look like a code.

File: recursion/recursion.md [hunk 8]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+     42
+     -----
+      6 ) 14 ( 2
+          12
+          -----
+           2
+
+Remainder is not yet zero, so we will now divide 6 by 2
+
+14 ) 48 ( 3
```
Comment (praveenscience) [inline recursion/recursion.md]: Enclose this inside <code>```</code> to make it look like a code.

File: recursion/recursion.md [hunk 9]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+          -----
+           2 ) 6 ( 3
+               6
+               ----
+               0
+
+Remainder is 0 so, we stop here
+
+Therefore, the required GCD is 2.
+
```
Comment (praveenscience) [inline recursion/recursion.md]: Get rid of empty lines.

File: recursion/recursion.md [hunk 10]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
 What do these function calls return?
 
 `fun1(1, 1)`
-`fun1(2, 1)`
+`fun1(2, 1)`
```
Comment (praveenscience) [inline recursion/recursion.md]: Remove trailing spaces.

File: recursion/recursion.md (hunk 1 above)
Comment (Kundan28) [reply recursion/recursion.md]: @Kumaranupam2020 Remove the white space after `[GCD]` and add white space before `of`.

File: recursion/recursion.md (hunk 10 above)
Comment (Kundan28) [inline recursion/recursion.md]: Whitespace at the end of the line 128 is not required. @Kumaranupam2020

Comment (praveenscience) [reply recursion/recursion.md]: Reported already. 😇

File: recursion/recursion.md [hunk 11]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+
+#### [Challenge] - GCD
+
+The [GCD] (https://en.wikipedia.org/wiki/Greatest_common_divisor)of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
+The greatest common divisor(GCD) is also known as:-
+* the greatest common factor (gcf),
+* highest common factor (hcf),
+* greatest common measure (gcm),
+* highest common divisor.
+
```
Comment (Kundan28) [inline recursion/recursion.md]: gcf -> GCF
hcf -> HCF
gcm -> GCM

File: recursion/recursion.md [hunk 12]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+
+For example:-
+
+the gcd of 48 and 14 is 2.
+
+### pseudocode to calculate gcd of two numbers:
```
Comment (Kundan28) [inline recursion/recursion.md]: pseudocode -> Pseudocode
gcd -> GCD

File: recursion/recursion.md [hunk 13]
```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
...
+               ----
+               0
+
+Remainder is 0 so, we stop here
+
+Therefore, the required GCD is 2.
+
+
+
+
```
Comment (Kundan28) [inline recursion/recursion.md]: Remove the blank lines.

File: recursion/recursion.md (hunk 10 above)
Comment (Kundan28) [reply recursion/recursion.md]: Oh. 😇

**Comment by praveenscience** [comment]
@Kumaranupam2020 please click on the Resolve Conversation for those you have fixed it.

File: recursion/recursion.md (hunk 11 above)
Comment (Kundan28) [reply recursion/recursion.md]: Great.

File: recursion/recursion.md (hunk 12 above)
Comment (Kundan28) [reply recursion/recursion.md]: @Kumaranupam2020 pseudocode -> Pseudocode

File: recursion/recursion.md (hunk 3 above)
Comment (Kumaranupam2020) [reply recursion/recursion.md]: please review it ..i tried to solve as u told

**Comment by Kundan28** [comment]
@Kumaranupam2020 Please try to address what you changed in the commit message instead of saying 
`Update recursion.md` everytime.

**Comment by Kundan28** [comment]
> @Kumaranupam2020 Please try to address what you changed in the commit message instead of saying
> `Update recursion.md` everytime.

And try to make all the asked changes in one commit for a single file.

File: recursion/recursion.md (hunk 6 above)
Comment (Kumaranupam2020) [reply recursion/recursion.md]: before which line?
 ### Check for understanding or Finding GCD of 48 & 14

Comment (Kundan28) [reply recursion/recursion.md]: @praveenscience ☝️

Comment (praveenscience) [reply recursion/recursion.md]: @Kumaranupam2020 Line 166 should be empty. 😄

File: recursion/recursion.md (hunk 3 above)
Comment (praveenscience) [reply recursion/recursion.md]: Looks good @Kumaranupam2020. Do it for others too, thanks.

File: recursion/recursion.md (hunk 6 above)
Comment (Kumaranupam2020) [reply recursion/recursion.md]: i think i have solved everything as requirred

Comment (praveenscience) [reply recursion/recursion.md]: Will check them... 😇

File: recursion/recursion.md [hunk 14]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+     -----
+      6 ) 14 ( 2
+          12
+          -----
+           2 ) 6 ( 3
+               6
+               ----
+               0
+```
+Remainder is 0 so, we stop here
```
Comment (praveenscience) [inline recursion/recursion.md]: Add a full stop.

File: recursion/recursion.md [hunk 15]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+          return x;
+      else
+          Call: GCD(y, x%y);
+      endif
+End
+```
+
+### Check for understanding
+
+Finding GCD of 48 & 14 :
```
Comment (praveenscience) [inline recursion/recursion.md]: Remove space before `:`

File: recursion/recursion.md [hunk 16]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+```
+14 ) 48 ( 3
+     42
+     -----
+      6
+```
+Remainder is not yet zero, so we will now divide 14 by 6
```
Comment (praveenscience) [inline recursion/recursion.md]: Add `:` at the end.

File: recursion/recursion.md [hunk 17]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+```
+14 ) 48 ( 3
+     42
+     -----
+      6 ) 14 ( 2
+          12
+          -----
+           2
+```
+Remainder is not yet zero, so we will now divide 6 by 2
```
Comment (praveenscience) [inline recursion/recursion.md]: Add `:` at the end.

**Comment by Kundan28** [comment]
@alodahl These changes are for https://github.com/Techtonica/curriculum/blob/master/recursion/recursion.md

**Comment by alodahl** [comment]
For the materials sections? That’s what #570 is about

**Comment by Kundan28** [comment]
> For the materials sections? That’s what #570 is about

Yeah.

**Comment by alodahl** [comment]
@Kundan28 I'm saying that I don't see changes for materials or supplemental resources here.

**Comment by Kundan28** [comment]
> @Kundan28 I'm saying that I don't see changes for materials or supplemental resources here.

Now I got what you meant to say.
@Kumaranupam2020 Please have a look to the @alodahl comment and make the required changes.

**Comment by Kumaranupam2020** [comment]
Okay i'll see it tomorrow.

**Comment by alodahl** [comment]
Can someone explain to me how this relates to the issue? How did so many people approve this?

**Comment by Kundan28** [comment]
@alodahl Should this PR be closed as the issue #570 is already closed.

File: recursion/recursion.md [hunk 18]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
 ```
 
 What do these function calls return?
-
 `fun1(1, 1)`
 `fun1(2, 1)`
 `fun1(2, 2)`
 `fun1(0, 2)`
+`fun1(0, 2)`
```
Comment (alodahl) [inline recursion/recursion.md]: Remove this duplicate

File: recursion/recursion.md [hunk 19]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+
+
+#### [Challenge] - GCD
+
+The [GCD](https://en.wikipedia.org/wiki/Greatest_common_divisor) of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
```
Comment (alodahl) [inline recursion/recursion.md]: Which are not all zero —> none of which are zero

File: recursion/recursion.md [hunk 20]
```diff
@@ -123,8 +123,72 @@ int fun1(int x, int y)
...
+Begin
+      if y = 0 then
+          return x;
+      else
+          Call: GCD(y, x%y);
+      endif
+End
+```
+
+### Check for understanding
```
Comment (alodahl) [inline recursion/recursion.md]: This section is not clear to me at all. I’m not sure it’s necessary. It seems better to just ask learners to solve the problem with your pseudo code in the challenge section.

**Comment by alodahl** [comment]
In the future, make sure your commit messages are all different, describing what you actually changed.  This will help you find your changes later if you need to,  and it is expected in a professional setting.

**Pull request merged**